hibernate-reactive-core/src/test/java/org/hibernate/reactive/OrderTest.java

```diff
@@ -0,0 +1,242 @@
+import jakarta.persistence.Id;
+import jakarta.persistence.metamodel.SingularAttribute;
+
+import static java.util.concurrent.TimeUnit.MINUTES;
+import static java.util.stream.Collectors.toList;
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.MYSQL;
+import static org.hibernate.reactive.containers.DatabaseConfiguration.DBType.MARIA;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+@DisabledIf( "skipDBs")
+@Timeout(value = 10, timeUnit = MINUTES)
+public class OrderTest extends BaseReactiveTest {
+	final Book book1 = new Book("9781932394153", "Hibernate in Action");
+	final Book book2 = new Book("9781617290459", "Java Persistence with Hibernate");
+
+	@Override
+	protected Collection<Class<?>> annotatedEntities() {
+		return List.of(
+				Book.class
+		);
+	}
+
+	static boolean skipDBs() {
+		// MySQL & MariaDB will fail because the simple 'String isbn' element will prevent book from being created correctly
+		// errorMessage=Table 'hreact.Book' doesn't exist, errorCode=1146
+		// NOTE: ORM's OrderTest will also fail with these DB's
+		return DatabaseConfiguration.dbType() == MYSQL ||
+				DatabaseConfiguration.dbType() == MARIA;
+	}
+
+	private CompletionStage<Void> populateDB() {
+		return getSessionFactory()
+				.withTransaction( session -> session.persist( book1, book2 ) );
+	}
+
+	private Uni<Void> populateDBMunity() {
+		return getMutinySessionFactory()
+				.withSession( session -> session.persistAll( book1, book2 )
+						.chain( session::flush ) );
+	}
+
+	@Test
+	public void testOrder(VertxTestContext context) {
+		test(
+				context,
+				populateDB().thenCompose( v -> getSessionFactory()
+						.withTransaction( (session, tx) -> session
+								.createSelectionQuery( "select title from Book", String.class )
+								.getResultList()
+								.thenAccept( list -> assertTrue( list.size() == 2 ) )
+								.thenCompose( vv -> {
+									MappingMetamodelImpl metamodel = (MappingMetamodelImpl) getSessionFactory().getMetamodel();
+									EntityDomainType<Book> bookType = metamodel.getJpaMetamodel()
+											.findEntityType( Book.class );
+									SingularAttribute<? super Book, ?> title = bookType.findSingularAttribute( "title" );
+									return session
+											.createSelectionQuery( "from Book", Book.class )
+											.setOrder( Order.asc( title ) )
+											.getResultList();
+								} ).thenAccept( books -> {
+									assertEquals( "Hibernate in Action", books.get( 0 ).title );
+									assertEquals( "Java Persistence with Hibernate", books.get( 1 ).title );
+								} )
+								.thenCompose( vv -> {
+									MappingMetamodelImpl metamodel = (MappingMetamodelImpl) getSessionFactory().getMetamodel();
+									EntityDomainType<Book> bookType = metamodel.getJpaMetamodel()
+											.findEntityType( Book.class );
+									SingularAttribute<? super Book, ?> title = bookType.findSingularAttribute( "title" );
```

Comment:
There's no reason to repeat these three lines everywhere: you can declare them once at the begging of the test